src/write_gcm.c

```diff
@@ -160,6 +160,72 @@ static void EVP_MD_CTX_free(EVP_MD_CTX *ctx) {
 }
 #endif
 
+// Multi-threading support for Curl and TLS. This is needed because wg_write
+// creates one thread for each queue (ATS and GSD).
+// See https://curl.haxx.se/libcurl/c/threadsafe.html
+
+#if OPENSSL_VERSION_NUMBER < 0x10003000L
+
+#include <openssl/crypto.h>
+
+static pthread_mutex_t *lockarray;
+
+static void lock_callback(int mode, int type, char *file, int line)
```

Comment:
What does `type` refer to? Isn't this just a lock number?

Reply:
Renamed to `index`